src/app/components/driver-list/driver-list.component.ts

```diff
@@ -106,84 +108,117 @@ export class DriverListComponent implements OnInit {
-                                  <td class="col">${results[0].duration.text}</td>
-                                  <td class="col">
-                                  <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#exampleModalCentered${element.id}"> View</button>
-                                    <div class="col-lg-5">
-                                     <div class="modal" id="exampleModalCentered${element.id}" tabindex="-1" role="dialog" aria-labelledby="exampleModalCenteredLabel" aria-hidden="true">
-                                      <div class="modal-dialog modal-dialog-centered" role="document">
-                                          <div class="modal-content">
-                                              <div class="modal-header">
-                                                  <h5 class="modal-title" id="exampleModalCenteredLabel">Contact Info:</h5>
-                                                  <button type="button" class="close" data-dismiss="modal" aria-label="Close">
-                                                     <span aria-hidden="true">×</span>
-                                                   </button>
-                                              </div>
-                                              <div class="modal-body">
-                                                  <h1>${name}</h1>
-                                                  <h3>Email: ${element.email}</h3>
-                                                  <h3>Phone: ${element.phone}</h3>
-                                              </div>
-                                              <div class="modal-footer">
-                                                <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
-                                              </div>
-                                            </div>
-                                         </div>
-                                       </div>
-                                  </div>
-                                  <div class="col-lg-6">
-                                      <div #maps id="gmap" class="img-responsive"></div>
-                                  </div>
-                                </td>
-                                <td class="col">${seats}</td>
-                                </tr>`;
+          if (status !== 'OK') {
+            alert('Error was: ' + status);
+          } 
+          else {
+            
+            // on success, for each driver, do this:
+          
+            /**
+             * 
+             * The architecture of Response is as following:
+             * 
+             * NOTE: 'rows' is an array where each row corresponds to an origin.
+             *        Since we are dealing with SINGLE origin, we can only have 
+             *        value at rows[0]. 
+             * 
+             * {
```

Comment:
The comment here of the structure is useful, but it would be a lot more useful if we defined an interface to get typing support.

- [ ] Define API response format in an interface. It doesn't need to include every single field, but at least the ones we will utilize.

- [ ] Provide the interface as the type for the response object.